Make ec2_group module not just fail

altruism opened the pull request:
A function was extracted without attention to local vs global variables. There was already a bugfix related to this but it didn't include the ec2 variable.

How to trigger this: Try to use the ec2_group module to create a new security group. The comments on #7592 are relevant. At that time, it was noticed that the `module` variable was not in scope in `get_target_from_rule` (https://github.com/ansible/ansible/issues/7592#issuecomment-45111472).

Example task:

```
- name: Create a security group
  local_action:
    module: ec2_group
    name: "i-am-a-security-group-that-does-not-exist-yet"
    description: a normal security group
```

You'll need to have AWS credentials set as environment variables, as described in the docs for this module, in order for this task as written to fail. N.B., the security group must not already exist in your AWS account, because if it does, the module will not attempt to create it and will not fail.

mpdehaan commented:
Can you please submit this with a description of how to trigger a failure, and also remove the merge commit (which we do not allow in history of pull requests)?   Use git pull --rebase vs git merge and git rebase vs git merge, and this will make this easier.  You should be able to cherry-pick your commits to a new branch, and also using a new branch for each pull request can make it easier to avoid merge commits as well.

Then we can look into merging this.

Thanks!

mpdehaan commented:
Also can you link the other bugfix number?   Thanks again!

altruism commented:
The bugfix was https://github.com/ansible/ansible/commit/b3bbca03d59c230212c9303d99952dad66eb082f#diff-317f4cc0ba915fbd7bd5ac28fdadc031 with issue #7592. I'm going to do the rebase thing right now.

altruism committed:
Add missing params to get_target_from_rule

get_target_from_rule was extracted from main() without enough parameters for
all the local variables it took with it from the extraction site.

altruism commented:
OK, rebase complete and description updated!  Is there anything else I need to do for this PR?

altruism commented:
@mpdehaan Did I rebase correctly?

jimi-c commented:
Merged, thanks!
